### DISCUSSION TITLE
Any plan to have unstyled version of Mantine

### DISCUSSION THREAD

**Discussion by kawpii** [mantinedev/mantine#748] [Q&A] [answered]
@rtivital Awesome work with Mantine! 

Is there unstyled Mantine components? I saw the [unstyled button](https://mantine.dev/core/button/#unstyled-button) do not know about others.
With unstyled mantine components I could use TailwindCSS to style things.

**Comment by rtivital** [comment, marked as answer]
Currently, I'm considering an option to add @mantine/headless package with unstyled components that will include some complex components. For other components like button, I do not think that it is a good idea, you can already overwrite all styles using Styles API.

**Reply by kawpii** [reply]
What does styles API use under the hood? (emotion, styled components etc)

**Reply by converter-user-old** [reply]
emotion see https://mantine.dev/theming/create-styles/

**Reply by kawpii** [reply]
@ruessej so the emotion runtime comes with the style API even though I used Mantine with TailwindCSS?

**Reply by rtivital** [reply]
Yes, Mantine always come with emotion

**Reply by Jonnotie** [reply]
Would love headless components so I can style it with Tailwind 🤤

**Reply by kawpii** [reply]
@Jonnotie Take look at [React-Creme]( https://github.com/prabhuignoto/react-creme/), Prime-React, [Carbon](https://github.com/carbon-design-system/carbon) which do not use CSS in JS, There is also Radix-UI which use Stiches

**Comment by kwiat1990** [comment]
That would be huge and would give a lot of welcomed flexibility.

**Reply by rtivital** [reply]
https://v5.mantine.dev/changelog/5-0-0/#unstyled-components

**Reply by kwiat1990** [reply]
@rtivital is there any specific plan regarding release of 5.0?

**Reply by ivan-kleshnin** [reply]
> All components now support unstyled prop which removes all non-essential Mantine styles from the component. This is useful when you want to style some component from scratch without overwriting any styles.

Will this prevent unnecessary bundling of default styles?

**Reply by rtivital** [reply]
No
